src/main/java/seedu/bookmark/model/wordstore/TagWordStore.java

```diff
@@ -0,0 +1,79 @@
+package seedu.bookmark.model.wordstore;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class TagWordStore extends WordStore {
+
+    private ArrayList<Word> tagWordStore;
```

Comment:
Maybe use a `HashMap` or `HashSet` for O(1) look ups?

Reply:
Thanks for the recommendation! I have included HashMap as an attribute. Thanks!